src/invisible/troubleshooting.md

```diff
@@ -21,15 +21,44 @@ Potential problems include:
 - Low storage space remaining on the Companion Device.
 - Unstable connection to the Pupil Invisible Glasses device during a recording.
 
-#### I accidentally updated my OnePlus 6 to an incompatible Android version!
+#### I accidentally updated my Companion device to an incompatible Android version!
 You need to rollback the Android version to be compatible again. Please note that all data on the phone will be lost during the rollback, so make backups accordingly.
 
-To execute the rollback, follow these steps:
+**OnePus 6**
+
+If you upgraded to Android 10 or newer and want to roll back, follow these steps:
 1. Download the ROM package [here](https://drive.google.com/file/d/1X8SVUH7UU7g0dcf-iWWXTPtMWzGguhUA/view?usp=sharing).
 1. Copy the ROM package to the mobile phone storage root directory.
 1. Go to Settings -> system updates -> top right corner icon -> local upgrade -> click on the installation package -> immediately upgrade -> system upgrade to 100%.
 1. Select Reboot system now.
- 
+
+**OnePlus 8/8T**
+
+If you upgraded to Android 12 or newer and want to roll back, follow these steps:
+1. Download the ROM package for your corresponding phone model to your computer
+
+   - [OnePlus 8](https://drive.google.com/file/d/1WtZ7bVwWPwYIGq4aWadeG-7MLu-KXhIS/view?usp=sharing)
+   - [OnePlus 8T](https://drive.google.com/file/d/1K2VI-R4gGN8mdS6FxLiRo48ICF_UBB7A/view?usp=sharing)
+
+2. Download the [official Rollback APK](https://oxygenos.oneplus.net/OPLocalUpdate_For_Android12.apk) to your computer
+3. Follow steps 1-5 from the [Transfer Exported Recordings to a Computer](/invisible/how-tos/data-collection-with-the-companion-app/transfer-recordings-via-usb.html#transfer-exported-recordings-to-a-computer) instructions
+4. Copy the downloaded ROM and APK to the top level of the device's Internal Storage (do not put it inside of any folders)
```

Comment:
> This process is the same for OnePlus 6 and 8 too.

Right, it is just to link a resource on how to copy files to the phone. How about the suggestion below
```suggestion
3. Copy the downloaded ROM and APK to the top level of the device's Internal Storage (do not put it inside of any folders; see steps 1-5 from the [Transfer Exported Recordings to a Computer](/invisible/how-tos/data-collection-with-the-companion-app/transfer-recordings-via-usb.html#transfer-exported-recordings-to-a-computer) instructions on how to connect your phone with your computer)
```

Alternatively, we coukd add this as a tip at the top of both rollback sections?

Reply:
Right, its not just about recording backup, but also about how to copy the ROM/APK to the phone. I'd be fine with either linking this at the top or like in the suggestion here. But something similar would then need to be mentioned for OnePlus 6.

Maybe it should more explicitly say "instructions on how to transfer files between your phone and your computer" rather than "how to connect them".

Reply:
Well, steps 1-5 are mostly about establishing the connection between phone and computer. But this is probably a technicality that is not relevant for the user. I will adjust the instructions for both phones.